src/database.js

```diff
@@ -36,7 +36,10 @@ function channel (path, event = 'value') {
   const channel = eventChannel(emit => {
     const callback = ref.on(
       event,
-      dataSnapshot => emit(dataSnapshot.val())
+      dataSnapshot => emit({
+        snapshot: dataSnapshot,
+        value: dataSnapshot.val()
```

Comment:
I think I'll keep `value` for now, the `data` example (did you mean [this one](https://firebase.google.com/docs/reference/js/firebase.database.DataSnapshot#val)?) is pretty small whereas `value` is similar to `snapshot.val()`.